docs/modules/reference/pages/configuration/history.adoc

```diff
@@ -0,0 +1,133 @@
+admin@opennms()> config:property-set elasticUrl http://es:9200
+admin@opennms()> config:update
+----
+
+. To make the alarm history feature start automatically on future service starts, add the line `opennms-alarm-history-elastic` to the file `$\{OPENNMS_HOME}/etc/featuresBoot.d/alarm.boot`.
+(Create the file if it does not exist.)
+. To make the alarm history feature start now, log in to the Karaf shell and run the following command to load the feature:
+
+ feature:install opennms-alarm-history-elastic
+
+[[ga-alarm-history-indexing]]
+== Alarm indexing
+
+When alarms are initially created, we push a document to Elasticsearch that includes all of the alarm fields as well as additional details on some of the related objects (i.e., the node.)
+
+In order to avoid pushing a new document every time a new event is reduced on to an existing alarm, we only push a new document when (at least) one of these conditions are met:
+
+. We have not recently pushed a document for that alarm. (See `alarmReindexDurationMs`.)
+. The severity of the alarm has changed.
+. The alarm has been acknowledged or unacknowledged.
+. Either of the associated sticky or journal memos have changed.
+. The state of the associated ticket has changed.
+. The alarm has been associated with, or removed, from a situation.
+. A related alarm has been added or removed from the situation.
+
+NOTE: To change this behavior and push a new document for *every* change, you can set `indexAllUpdates` to `true`.
+
+When alarms are deleted, we push a new document that contains the alarm id, reduction key, and deletion time.
+
+The following table describes a subset of the fields in the alarm document:
+
+[options="header"]
+[cols="1,3"]
+|===
+| Field | Description
+
+| @first_event_time
+| Timestamp in milliseconds associated with the first event that triggered this alarm.
+
+| @first_event_time
+| Timestamp in milliseconds associated with the last event that triggered this alarm.
+
+| @update_time
+| Timestamp in milliseconds at which the document was created.
+
+| @deleted_time
+| Timestamp in milliseconds when the alarm was deleted.
+
+| id
+| Database ID associated with the alarm.
+
+| reduction_key
+| Key used to reduce events on to the alarm.
+
+| severity_label
+| Severity of the alarm.
+
+| severity_id
+| Numerical ID used to represent the severity.
+
+|===
+
+[[ga-alarm-history-options]]
+== Options
+
+In addition to those mentioned in <<elasticsearch/introduction.adoc#ga-elasticsearch-integration-configuration, Elasticsearch Integration Configuration>>, you can set the following optional properties in `$\{OPENNMS_HOME}/etc/org.opennms.features.alarms.history.elastic.cfg`:
+
+[options="header"]
+[cols="1,3,2"]
+|===
+| Property
+| Description
+| Default
+
+| indexAllUpdates
+| Index every alarm update, including simple event reductions.
+| false
+
+| alarmReindexDurationMs
+| Number of milliseconds to wait before re-indexing an alarm if nothing "interesting" has changed.
```

Comment:
```suggestion
| Number of milliseconds to wait before re-indexing an alarm if nothing "interesting" has changed.
```
"interesting" is vague. Could we say "if nothing significant (like x or y) has changed." and then list two items. Or does this refer back to the list that we had earlier in this section?